screens/CreateEventScreen.jsx

```diff
@@ -0,0 +1,414 @@
+import React, { useState } from 'react';
+import { View, Picker, FlatList, Switch, Text, TextInput, Alert, ScrollView, Keyboard ,StyleSheet, SafeAreaView} from 'react-native';
+import { Ionicons } from '@expo/vector-icons';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+import { createEvent } from '../api/firebaseMethods';
+import * as ImagePicker from 'expo-image-picker';
+import * as ImageManipulator from 'expo-image-manipulator';
+import DefaultImg from '../assets/images/defaultprofile.png'
+import * as firebase from 'firebase';
+import { editProfile, uploadPicture } from '../api/firebaseMethods';
+import { follow, unfollow } from '../api/firebaseMethods'
+import { useUser, useUserUpdate } from '../hooks/UserContext';
+
+export default function CreateEventScreen({ navigation }) {
+  const [wordCount, setWordCount] = useState(0)
+
+  const [showName, setShowName] = useState('');
+  const [description, setDescription] = useState('');
+  const [tags, setTags] = useState([])
+
+  const [ticketPrice, setTicketPrice] = useState('');
+  const [ticketLink, setTicketLink] = useState('');
+  const [intermission, setIntermission] = useState(false);
+  const toggleIntermission = () => setIntermission(previousState => !previousState);
+  const [ticketRequired, setTicketRequired] = useState(false);
+  const toggleTicketRequired = () => setTicketRequired(previousState => !previousState);
+
+  const [venueName, setVenueName] = useState('');
+  const [venueAddress, setVenueAddress] = useState('');
+  const [wheelchair, setWheelchair] = useState(false)
+  const toggleWheelchair = () => setWheelchair(previousState => !previousState);
+  const [hearingAssistance, setHearingAssistance] = useState(false);
+  const toggleHearingAssistance = () => setHearingAssistance(previousState => !previousState);
+
+  const [cast, setCast] = useState([{name:'', role:''}]);
+  var castId = 0;
```

Comment:
hmmm... im thinking we can get rid of this to allow flexible editing like you mentioned in your todo. what if the cast/creatives section was a flatlist that converts the cast data in the state to pairs of name & role fields (Cast.jsx objects)? you can use the index of the item in the flatlist to specify changes.